File: 프로그래머스/lv2/12945. 피보나치 수/피보나치 수.swift
func solution(_ n:Int) -> Int {
    var pre = 0
    var curr = 1
    var sum = 1
    
    for i in 2...n {
        sum = pre + curr % 1234567
        pre = curr % 1234567
        curr = sum
    }
    
    return sum % 1234567
}

File: 프로그래머스/lv2/12945. 피보나치 수/피보나치 수.js
function solution(n) {
    let result = 1;
    let pre = 1;
    
    for(let i = 2; i < n; i++) {
        const temp = result;
        result = (pre+result)%1234567;
        pre = temp;
    }
    
    return result;
}